pandas/core/indexes/interval.py

```diff
@@ -97,6 +98,27 @@ def _new_IntervalIndex(cls, d):
     return cls.from_arrays(**d)
 
 
+class setop_check:
+
```

Comment:
can you add a doc-string here (also pls add type annotations where easy / appropriate) if you can (generally we want to do this).